From 8068a28350a65c738279a6a27052513087e69619 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 5 Apr 2023 11:57:44 +1000
Subject: [PATCH 1/8] do not package third_party folder

---
 python/setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: triton/python/setup.py
===================================================================
--- triton.orig/python/setup.py	2024-04-01 16:58:26.077448738 -0500
+++ triton/python/setup.py	2024-04-02 12:25:45.247980479 -0500
@@ -360,7 +360,6 @@
         "triton/ops/blocksparse",
         "triton/runtime",
         "triton/runtime/backends",
-        "triton/third_party",
         "triton/tools",
     ],
     install_requires=["filelock"],
Index: triton/python/MANIFEST.in
===================================================================
--- triton.orig/python/MANIFEST.in	2024-04-02 12:24:58.804145806 -0500
+++ triton/python/MANIFEST.in	2024-04-02 12:25:52.665351813 -0500
@@ -1,5 +1,4 @@
 graft src
-graft triton/third_party
 graft triton/tools
 graft triton/runtime/backends/
 graft triton/language/extra
