--- a/build_artifacts/triton_1680650972620/work/bin/CMakeLists.txt	2023-03-03 09:57:45.000000000 +1000
+++ b/build_artifacts/triton_1680650972620/work/bin/CMakeLists.txt	2023-04-05 09:48:45.122386723 +1000
@@ -1,4 +1,4 @@
-add_subdirectory(FileCheck)
+# add_subdirectory(FileCheck)
 # add_llvm_executable(FileCheck FileCheck/FileCheck.cpp)
 # target_link_libraries(FileCheck PRIVATE LLVMFileCheck LLVMSupport)
 
