From 5dd75f7a49550ca5c461e1ab1c55d9a1ccda997d Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:49:23 +1000
Subject: [PATCH 2/8] Fix error that FileCheck is already defined

---
 bin/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/CMakeLists.txt b/bin/CMakeLists.txt
index 906f635f8..c6b419782 100644
--- a/bin/CMakeLists.txt
+++ b/bin/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_subdirectory(FileCheck)
+# add_subdirectory(FileCheck)
 # add_llvm_executable(FileCheck FileCheck/FileCheck.cpp)
 # target_link_libraries(FileCheck PRIVATE LLVMFileCheck LLVMSupport)
 
