From 30c2df0dc9a05ab5a251aeff4a473dffebd55a5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@quansight.com>
Date: Fri, 1 Aug 2025 13:31:58 +0200
Subject: [PATCH 3/4] Use system PATH to find tools (in CONDA_PREFIX)

---
 python/triton/knobs.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/python/triton/knobs.py b/python/triton/knobs.py
index cd61f3972..e51566548 100644
--- a/python/triton/knobs.py
+++ b/python/triton/knobs.py
@@ -4,6 +4,7 @@ import functools
 import importlib
 import os
 import re
+import shutil
 import subprocess
 import sysconfig
 import pathlib
@@ -205,12 +206,11 @@ class env_nvidia_tool(env_base[str, NvidiaTool]):
     def transform(self, path: str) -> NvidiaTool:
         # We still add default as fallback in case the pointed binary isn't
         # accessible.
-        if path is not None:
-            paths = [path, self.default_path]
-        else:
-            paths = [self.default_path]
+        paths = [path, self.default_path, shutil.which(self.binary)]
 
         for path in paths:
+            if path is None:
+                continue
             if tool := NvidiaTool.from_path(path):
                 return tool
 
-- 
2.53.0

