From 206c8923442f6525e419395f9ea82df347f8cf04 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 25 Feb 2022 14:59:48 +1100
Subject: [PATCH 3/8] properly point to triton includes

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0d361fc7..378a1150c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,7 @@ if(TRITON_BUILD_PYTHON_MODULE)
     message(STATUS "Adding Python module")
     set(PYTHON_SRC_PATH ${CMAKE_CURRENT_SOURCE_DIR}/python/src)
     set(PYTHON_SRC ${PYTHON_SRC_PATH}/main.cc ${PYTHON_SRC_PATH}/triton.cc)
-    include_directories("." ${PYTHON_SRC_PATH})
+    include_directories("./include" ${PYTHON_SRC_PATH})
     if (PYTHON_INCLUDE_DIRS)
       include_directories(${PYTHON_INCLUDE_DIRS})
     else()
