From 0c8b2725a78a081859b9890b7475644ed3f6adf4 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:53:42 +1000
Subject: [PATCH 5/8] Unvendor third party libs

---
 CMakeLists.txt  |  3 ++-
 python/setup.py | 11 ++++++-----
 2 files changed, 8 insertions(+), 6 deletions(-)

Index: triton/CMakeLists.txt
===================================================================
--- triton.orig/CMakeLists.txt	2024-05-31 11:26:58.261671000 -0500
+++ triton/CMakeLists.txt	2024-05-31 11:37:42.682379042 -0500
@@ -45,7 +45,8 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 # Third-party
-include_directories(${PYBIND11_INCLUDE_DIR})
+# include_directories(${PYBIND11_INCLUDE_DIR})
+find_package(pybind11 REQUIRED)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -D__STDC_FORMAT_MACROS  -fPIC -std=gnu++17 -fvisibility=hidden -fvisibility-inlines-hidden")
 
Index: triton/python/setup.py
===================================================================
--- triton.orig/python/setup.py	2024-05-31 11:37:37.626430157 -0500
+++ triton/python/setup.py	2024-05-31 11:38:00.788446000 -0500
@@ -105,7 +105,7 @@
 
 
 def get_thirdparty_packages(triton_cache_path):
-    packages = [get_pybind11_package_info(), get_llvm_package_info()]
+    packages = [get_llvm_package_info()]
     thirdparty_cmake_args = []
     for p in packages:
         package_root_dir = os.path.join(triton_cache_path, p.package)
@@ -319,27 +319,6 @@
         subprocess.check_call(["cmake", "--build", ".", "--target", "mlir-doc"], cwd=cmake_dir)
 
 
-download_and_copy(
-    src_path="bin/ptxas",
-    variable="TRITON_PTXAS_PATH",
-    version="12.3.52",
-    url_func=lambda arch, version:
-    f"https://anaconda.org/nvidia/cuda-nvcc/12.3.52/download/linux-{arch}/cuda-nvcc-{version}-0.tar.bz2",
-)
-download_and_copy(
-    src_path="bin/cuobjdump",
-    variable="TRITON_CUOBJDUMP_PATH",
-    version="12.3.52",
-    url_func=lambda arch, version:
-    f"https://anaconda.org/nvidia/cuda-cuobjdump/12.3.52/download/linux-{arch}/cuda-cuobjdump-{version}-0.tar.bz2",
-)
-download_and_copy(
-    src_path="bin/nvdisasm",
-    variable="TRITON_NVDISASM_PATH",
-    version="12.3.52",
-    url_func=lambda arch, version:
-    f"https://anaconda.org/nvidia/cuda-nvdisasm/12.3.52/download/linux-{arch}/cuda-nvdisasm-{version}-0.tar.bz2",
-)
 
 setup(
     name=os.environ.get("TRITON_WHEEL_NAME", "triton"),
Index: triton/include/triton/Target/PTX/TmaMetadata.h
===================================================================
--- triton.orig/include/triton/Target/PTX/TmaMetadata.h	2024-05-31 11:26:58.261838000 -0500
+++ triton/include/triton/Target/PTX/TmaMetadata.h	2024-05-31 11:37:42.683789027 -0500
@@ -24,7 +24,7 @@
 #ifndef TRITON_TARGET_PTX_TMAMETADATA_H
 #define TRITON_TARGET_PTX_TMAMETADATA_H
 
-#include "python/triton/third_party/cuda/include/cuda.h"
+#include "cuda.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/Debug.h"
 #include "llvm/Support/Format.h"
