From 76c0f03874787fd71c63b5cdd40084d106007d9b Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:54:11 +1000
Subject: [PATCH 6/8] Avoid pip check error as cmake is not a python module

---
 python/setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index e99b38408..4174506dc 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -212,7 +212,6 @@ setup(
     long_description="",
     packages=["triton", "triton/_C", "triton/language", "triton/tools", "triton/impl", "triton/ops", "triton/runtime", "triton/ops/blocksparse"],
     install_requires=[
-        "cmake",
         "filelock",
         "torch",
         "lit",
