From ffbc365c692aad26ade6136758fa0d1710db95f7 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:54:48 +1000
Subject: [PATCH 7/8] Fix TableGen issues

---
 CMakeLists.txt | 31 ++++++++++++++++++++++++++++---
 1 file changed, 28 insertions(+), 3 deletions(-)

Index: triton/CMakeLists.txt
===================================================================
--- triton.orig/CMakeLists.txt	2024-04-02 12:41:48.288157839 -0500
+++ triton/CMakeLists.txt	2024-04-02 12:42:38.692262266 -0500
@@ -68,7 +68,7 @@
         AMDGPUInfo AMDGPUcodegen
       )
     else()
-      find_package(LLVM 11 REQUIRED COMPONENTS "nvptx;amdgpu")
+      find_package(LLVM REQUIRED COMPONENTS "nvptx;amdgpu")
     endif()
 
     message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
@@ -172,7 +172,7 @@
 # endif()
 
 # MLIR
-find_package(MLIR REQUIRED CONFIG PATHS ${MLIR_DIR})
+find_package(MLIR REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH "${MLIR_CMAKE_DIR}")
 list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
@@ -181,6 +181,31 @@
 include(AddLLVM)
 include(AddMLIR)
 
+find_program(MLIR_TABLEGEN_EXE mlir-tblgen)
+if(NOT MLIR_TABLEGEN_EXE)
+  message(FATAL_ERROR "mlir-tblgen not found. Please ensure it is in your $PATH.")
+endif()
+
+# Create a custom target for mlir-tblgen
+add_custom_target(mlir-tblgen-wrapper
+  COMMAND ${MLIR_TABLEGEN_EXE} ${ARGN}
+  COMMENT "Running mlir-tblgen-wrapper"
+  VERBATIM
+)
+
+function(mlir_tablegen ofn)
+  set(output_file "${CMAKE_CURRENT_BINARY_DIR}/${ofn}")
+  add_custom_command(OUTPUT ${output_file}
+                     COMMAND ${MLIR_TABLEGEN_EXE} ${LLVM_TARGET_DEFINITIONS} ${ARGN} -I ${LLVM_INCLUDE_DIRS} -I ${MLIR_INCLUDE_DIRS} -I ${CMAKE_SOURCE_DIR}/include -o ${output_file}
+                     DEPENDS mlir-tblgen-wrapper
+                     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                     COMMENT "Generating ${output_file}"
+                     VERBATIM)
+  set(TABLEGEN_OUTPUT ${TABLEGEN_OUTPUT} ${output_file} PARENT_SCOPE)
+  include_directories(${CMAKE_CURRENT_BINARY_DIR})
+endfunction()
+
+
 # Disable warnings that show up in external code (gtest;pybind11)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-covered-switch-default")
 
