From e50454a16bafa90bcd028b2eb6a6a173bd6e15db Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 7 May 2023 14:55:38 +1000
Subject: [PATCH 8/8] Search for libs in CONDA_PREFIX instead of third_party
 directory

Index: triton/python/triton/common/build.py
===================================================================
--- triton.orig/python/triton/common/build.py	2024-06-04 16:32:22.918259000 -0500
+++ triton/python/triton/common/build.py	2024-06-04 16:36:01.615996127 -0500
@@ -59,7 +59,7 @@
 @functools.lru_cache()
 def cuda_include_dir():
     base_dir = os.path.join(os.path.dirname(__file__), os.path.pardir)
-    cuda_path = os.path.join(base_dir, "third_party", "cuda")
+    cuda_path = os.path.join(os.environ['CONDA_PREFIX'])
     return os.path.join(cuda_path, "include")
 
 
Index: triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp
===================================================================
--- triton.orig/lib/Target/LLVMIR/LLVMIRTranslation.cpp	2024-06-04 16:32:22.918374000 -0500
+++ triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp	2024-06-04 16:36:01.617095623 -0500
@@ -286,35 +286,21 @@
       externLibs.try_emplace(libdevice, env_path);
       return externLibs;
     }
-    // Search for libdevice relative to its library path if used from Python
-    // Then native code is in `triton/_C/libtriton.so` and libdevice in
-    // `triton/third_party/cuda/lib/libdevice.10.bc`
-    static const auto this_library_path = getThisLibraryPath();
-    static const auto runtime_path =
-        this_library_path.parent_path().parent_path() / "third_party" / "cuda" /
-        "lib" / "libdevice.10.bc";
-    if (fs::exists(runtime_path)) {
-      externLibs.try_emplace(libdevice, runtime_path.string());
+    // Search for libdevice in CONDA_PREFIX
+    const char* conda_prefix_cstr = std::getenv("CONDA_PREFIX");
+    if (!conda_prefix_cstr) {
+        llvm::report_fatal_error("CONDA_PREFIX environment variable not set.");
+    }
+    std::string conda_prefix(conda_prefix_cstr);
+    fs::path conda_prefix_path(conda_prefix);
+
+    static const auto libdevice_path = conda_prefix_path / "nvvm" / "libdevice" / "libdevice.10.bc";
+    if (fs::exists(libdevice_path)) {
+      externLibs.try_emplace(libdevice, libdevice_path.string());
     } else {
-      // When using the Math Dialect, it is possible that some ops (e.g., log)
-      // are lowered to a function call. In this case, we need to link libdevice
-      // using its default path:
-      // [triton root dir]/python/triton/language/libdevice.10.bc
-      // TODO(Keren): handle external linkage other than libdevice?
-      static const auto this_file_path = std::filesystem::path(__FILE__);
-      static const auto compiletime_path = this_file_path.parent_path()
-                                               .parent_path()
-                                               .parent_path()
-                                               .parent_path() /
-                                           "python" / "triton" / "third_party" /
-                                           "cuda" / "lib" / "libdevice.10.bc";
-      if (!fs::exists(compiletime_path)) {
-        std::string error_msg = "Can't find libdevice at neither " +
-                                runtime_path.string() + " nor " +
-                                compiletime_path.string();
+        std::string error_msg = "Can't find libdevice.10.bc at " +
+                                libdevice_path.string();
         llvm::report_fatal_error(error_msg.c_str());
-      }
-      externLibs.try_emplace(libdevice, compiletime_path.string());
     }
   }
 
Index: triton/python/triton/common/backend.py
===================================================================
--- triton.orig/python/triton/common/backend.py	2024-06-04 16:32:22.918317000 -0500
+++ triton/python/triton/common/backend.py	2024-06-04 16:37:14.064998559 -0500
@@ -93,6 +93,8 @@
 
 
 def get_backend(device_type: str):
+    if device_type == "cuda":
+        return None
     if device_type not in _backends:
         device_backend_package_name = f"...third_party.{device_type}"
         if importlib.util.find_spec(device_backend_package_name, package=__spec__.name):
@@ -109,7 +111,7 @@
     base_dir = os.path.join(os.path.dirname(__file__), os.pardir)
     paths = [
         os.environ.get(f"TRITON_{binary.upper()}_PATH", ""),
-        os.path.join(base_dir, "third_party", "cuda", "bin", binary)
+        os.path.join(os.environ["CONDA_PREFIX"], "bin", binary)
     ]
 
     for p in paths:
Index: triton/python/triton/language/math.py
===================================================================
--- triton.orig/python/triton/language/math.py	2024-06-04 16:32:22.918181000 -0500
+++ triton/python/triton/language/math.py	2024-06-04 16:36:01.618333161 -0500
@@ -11,7 +11,7 @@
     if is_hip():
         default = os.path.join(third_party_dir, "hip", "lib", "bitcode", "cuda2gcn.bc")
     else:
-        default = os.path.join(third_party_dir, "cuda", "lib", "libdevice.10.bc")
+        default = os.path.join(os.environ["CONDA_PREFIX"], "nvvm", "libdevice", "libdevice.10.bc")
 
     return os.getenv("TRITON_LIBDEVICE_PATH", default)
 
Index: triton/python/triton/tools/build_extern.py
===================================================================
--- triton.orig/python/triton/tools/build_extern.py	2024-06-04 16:32:22.918072000 -0500
+++ triton/python/triton/tools/build_extern.py	2024-06-04 16:36:01.618837868 -0500
@@ -277,7 +277,7 @@
         header_str += "    import torch\n"
         header_str += "    third_party_dir =  os.path.join(os.path.dirname(os.path.abspath(__file__)), \"..\", \"third_party\")\n"
         header_str += "    if torch.version.hip is None:\n"
-        header_str += "        default = os.path.join(third_party_dir, \"cuda\", \"lib\", \"libdevice.10.bc\")\n"
+        header_str += "        default = os.path.join(os.path.dirname(os.environ[\"CONDA_PREFIX\"], \"nvvm\", \"libdevice\", \"libdevice.10.bc\")"
         header_str += "    else:\n"
         header_str += "        default = ''\n"
         header_str += "    return os.getenv(\"TRITON_LIBDEVICE_PATH\", default)\n"
